jaxmarl/utils/snd.py

```diff
@@ -87,16 +108,22 @@ def distance_vector(agent_i):
         corresponding observations, add to flatten into a vector containing the added distances
         between i and each other agent 
         """
-        qvals_i = get_policy_outputs(agent_i) # [timesteps, batch_size, n_agents, action_dim]
-
-        # convert qvals to categorical distributions
-        qval_maxs = jnp.max(qvals_i, axis=-1, keepdims=True)
-        qvals_i = qvals_i - qval_maxs
-        categorical_i = jnp.exp(qvals_i) / jnp.sum(jnp.exp(qvals_i), axis=-1, keepdims=True)
+        if policy == 'qmix':
+            qvals_i = get_policy_outputs(agent_i) # [timesteps, batch_size, n_agents, action_dim]
+
+            # convert qvals to categorical distributions
+            qval_maxs = jnp.max(qvals_i, axis=-1, keepdims=True)
+            qvals_i = qvals_i - qval_maxs
+            categorical_i = jnp.exp(qvals_i) / jnp.sum(jnp.exp(qvals_i), axis=-1, keepdims=True)
+        elif policy == 'mappo':
+            categorical_i = get_policy_outputs(agent_i)
+        else:
+            categorical_i = get_policy_outputs(agent_i)
```

Comment:
Does this work now with DAgger? (I implemented DAgger as if it was greedy-value-based so I'm not sure if defaulting to sampling from policy will work)

Reply:
when I added it to dagger I just called it as if I was calling it from qmix since the forward pass is the same so it should be fine

Reply:
Got it thanks!